client/src/app/core/ui-services/personal-note.service.ts

```diff
@@ -66,4 +66,38 @@ export class PersonalNoteService {
             await this.http.put(`rest/users/personal-note/${pnObject.id}/`, pnObject);
         }
     }
+
+    /**
+     * Sets the 'favorite' status for several models of a type in bulk
+     *
+     * @param models The model the content belongs to
+     * @param star The new 'favorite' status
+     */
+    public async bulkSetStar(models: BaseModel[] | BaseViewModel[], star: boolean): Promise<void> {
+        if (!models.length) {
+            throw new Error('No models selected');
+        }
+        const pnObject: Partial<PersonalNoteObject> = this.personalNoteObject || {};
+        if (!pnObject.notes) {
+            pnObject.notes = {};
+        }
+        const colString = models[0].collectionString;
+        if (!pnObject.notes[colString]) {
+            pnObject.notes[colString] = {};
+        }
+        for (const model of models) {
+            if (!star && pnObject.notes[colString][model.id] && pnObject.notes[colString][model.id].star) {
+                pnObject.notes[colString][model.id].star = false;
+            } else if (star && !pnObject.notes[colString][model.id]) {
+                pnObject.notes[colString][model.id] = { star: true, note: '' };
+            } else if (star && !pnObject.notes[colString][model.id].star) {
+                pnObject.notes[colString][model.id].star = true;
+            }
+        }
+        if (!pnObject.id) {
```

Comment:
Please make a `private async savePersonalNoteObject(personalNoteObject: PersonalNoteObject): Promise<void>` out of this, because it is udes multiple times